src/Test/Utilities/Shared/Syntax/SourceUtilities.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Roslyn.Test.Utilities.Syntax
+{
+    internal sealed class RandomizedSourceText: SourceText
+    {
+        private char hash(int position)
+        {
+            // good enough "random looking" hasher via the xorshift128plus algorithm.
+            UInt64 state0 = (UInt64) 1;
+            UInt64 state1 = (UInt64) position;
+            unchecked
+            {
+                var s1 = state0;
+                var s0 = state1;
+                state0 = s0;
+                s1 ^= s1 << 23;
+                s1 ^= s1 >> 17;
+                s1 ^= s0;
+                s1 ^= s0 >> 26;
+                state1 = s1;
+                return (char)(state1 + s0);
+            }
+        }
+
+        public override char this[int position] => hash(position);
+
+        public override Encoding Encoding => Encoding.UTF8;
+
+        public override int Length
+        {
+            get
+            {
+                return 40 * 1000 * 1000;
+            }
+        }
+
+        public override void CopyTo(int sourceIndex, char[] destination, int destinationIndex, int count)
+        {
+            for (var i = 0; i < count; i++)
+            {
+                destination[destinationIndex + i] = hash(sourceIndex);
```

Comment:
> `sourceIndex` is never incremented

Fixed, thanks.

> Also, should we throw throw on an attempt to get characters out of range?

I doubt this would improve test coverage of that case, but adding it would be easy enough.